Proposed changes to Repository filtering doc (#1105)
ezsystems/developer-documentation: docs/api/public_php_api_search.md

For example, the following command lists all Content items under the specified parent Location
and sorts them in descending order by name:

> ``` php hl_lines="15-18"

Reviewer: ```suggestion
``` php hl_lines="15 16 17 18"
```

        }

        return self::SUCCESS;
    }
}
```

The same Filter can be applied to find Locations instead of Content items, for example:

``` php hl_lines="19"
// ...
use eZ\Publish\API\Repository\LocationService;
use eZ\Publish\API\Repository\Values\Filter\Filter;
use eZ\Publish\API\Repository\Values\Content\Query;
use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
use eZ\Publish\API\Repository\Values\Content\Query\SortClause;

class FilterCommand extends Command
{
    // ...
    protected function execute(InputInterface $input, OutputInterface $output): int
    {
        $parentLocationId = (int)$input->getArgument('parent-location-id');
        $filter = new Filter();
        $filter
            ->withCriterion(new Criterion\ParentLocationId($parentLocationId))
            ->withSortClause(new SortClause\ContentName(Query::SORT_DESC));

        $result = $this->locationService->find($filter, []);

        $output->writeln('Found ' . $result->totalCount . ' items');

        foreach ($result as $location) {
            $output->writeln($location->getContent()->getName());
        }

        return self::SUCCESS;
    }
}
```

> Notice that the total number of items is retrieved differently depending on whether we're dealing
> with `ContentList` or `LocationList`. 

Reviewer: ```suggestion
Notice that the total number of items is retrieved differently for `ContentList` and `LocationList`. 
```

!!! caution

>     The total count is **a total number of matched items**, regardless of pagination settings.

Reviewer: ```suggestion
    The total count is the total number of matched items, regardless of pagination settings.
```

> !!! tip "Repository Filtering is SiteAccess-aware."

Reviewer: ```suggestion
!!! tip "Repository filtering is SiteAccess-aware"
```

>     Repository Filtering is SiteAccess-aware, which means that the second argument of its respective
>     `find` methods can be skipped. In that case languages from a current context are injected and
>     added as a LanguageCode Criterion filter. 

Reviewer: ```suggestion
    Repository filtering is SiteAccess-aware, which means you can skip the second argument the `find` methods.
    In that case languages from a current context are injected and added as a LanguageCode Criterion filter. 
```

!!! note "Search Criteria and Sort Clause availability"

    Not all Search Criteria and Sort Clauses are available for use in Repository filtering.
    
    Only Criteria implementing [`FilteringCriterion`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringCriterion.php)
    and Sort Clauses implementing [`FilteringSortClause`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringSortClause.php)
    are supported.

    See to [Search Criteria](../guide/search/search_criteria_reference.md)
    and [Sort Clause reference](../guide/search/sort_clause_reference.md) for details.
    
!!! tip

>     It's recommended to use an IDE capable of recognizing type hints when working with Repository Filtering.
>     In case of using unsupported Criterion or Sort Clause, such IDE will immediately hint an issue. 

Reviewer: ```suggestion
    It is recommended to use an IDE that can recognize type hints when working with Repository Filtering.
    If you try to use an unsupported Criterion or Sort Clause, the IDE will indicate an issue. 
```
